pkg/connector/database_users.go

```diff
@@ -151,6 +160,45 @@ func (o *databaseUserBuilder) CreateAccount(ctx context.Context, accountInfo *v2
 		}
 	}
 
+	l.Info("creating database user", zap.String("userId", userId))
+
+	var user atlasUserResponse
+	if userId != "" {
+		user, _, err = o.client.MongoDBCloudUsersApi.GetOrganizationUser(ctx, orgId, userId).Execute() //nolint:bodyclose // The SDK handles closing the response body
+		if err != nil {
+			return nil, nil, nil, fmt.Errorf("failed to get user by id: %w", err)
+		}
+	} else {
+		response, _, err := o.client.MongoDBCloudUsersApi.ListOrganizationUsers(ctx, orgId).Username(email).Execute() //nolint:bodyclose // The SDK handles closing the response body
+		if err != nil {
+			if atlasErr, ok := admin.AsError(err); ok {
+				switch atlasErr.ErrorCode {
+				case "CANNOT_ADD_PENDING_USER":
```

Comment:
nit: can you wrap with `status.Code` (`codes.PermissionDenied`, or whatever is valid for these errors)

Reply:
Added gprc status